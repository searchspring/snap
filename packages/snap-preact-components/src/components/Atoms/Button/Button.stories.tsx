import { h } from 'preact';

import { Button } from './Button';

import { componentArgs } from '../../../utilities';

export default {
	title: `Atoms/Button`,
	component: Button,
	argTypes: {
		content: {
			description: 'Content to be displayed in button',
			table: {
				type: {
					summary: 'string, JSX',
				},
			},
			control: { type: 'text' },
		},
		children: {
			description: 'Content to be displayed in button (using children)',
			table: {
				type: {
					summary: 'string, JSX',
				},
			},
		},
		disabled: {
			defaultValue: false,
			description: 'Disable button',
			table: {
				type: {
					summary: 'boolean',
				},
				defaultValue: { summary: false },
			},
			control: { type: 'boolean' },
		},
		onClick: {
			description: 'Button click event handler',
			table: {
				type: {
					summary: 'function',
				},
			},
			action: 'onClick',
		},
		color: {
			description: 'Button color',
			table: {
				type: {
					summary: 'string',
				},
				defaultValue: { summary: '#333' },
			},
			control: { type: 'color' },
		},
		backgroundColor: {
			description: 'Button background color',
			table: {
				type: {
					summary: 'string',
				},
			},
			control: { type: 'color' },
		},
		borderColor: {
			description: 'Button border color',
			table: {
				type: {
					summary: 'string',
				},
				defaultValue: { summary: '#333' },
			},
			control: { type: 'color' },
		},
		native: {
			defaultValue: false,
			description: 'Render as unstyled native button',
			table: {
				type: {
					summary: 'boolean',
				},
				defaultValue: { summary: false },
			},
			control: { type: 'boolean' },
		},
		...componentArgs,
	},
};

const Template = (args) => <Button {...args} />;

export const Default = Template.bind({});
Default.args = {
	content: 'button',
};

export const Disabled = Template.bind({});
Disabled.args = {
	content: 'disabled',
	disabled: true,
};

export const Native = Template.bind({});
Native.args = {
	content: 'native',
	native: true,
};
