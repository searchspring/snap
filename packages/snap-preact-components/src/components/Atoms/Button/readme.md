## Button

Render as unstyled native button, or a div styled like a button based on the optional native prop. can take content or children, styling props available, has the ability to change the onClick functionality. 

## Usage

Content prop

``` jsx
<Button content={"click me!"} onClick={() => somefunc}/>
```
 
Children

``` jsx
<Button onClick={() => somefunc}>
       click me!
</Button>
```
