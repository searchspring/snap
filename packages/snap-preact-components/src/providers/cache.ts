export { CacheProvider } from '@emotion/react';
import createCache from '@emotion/cache';

export const cache = createCache({
	key: 'ss',
	prepend: true,
});
