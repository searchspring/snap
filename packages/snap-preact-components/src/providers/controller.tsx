import { jsx } from '@emotion/react';
import preact, { h, createContext, ComponentChildren, ComponentType } from 'preact';
import { useContext } from 'preact/hooks';
import type { AbstractController } from '@searchspring/snap-controller';

const ControllerContext = createContext<null | AbstractController>(null);

export const ControllerProvider = ({ children, controller }: { children: ComponentChildren; controller: AbstractController }) => {
	return <ControllerContext.Provider value={controller}>{children}</ControllerContext.Provider>;
};

export const useController = () => useContext(ControllerContext);

export const withController = (Component: ComponentType) => (props: any) => {
	return <Component controller={useController()} {...props} />;
};
