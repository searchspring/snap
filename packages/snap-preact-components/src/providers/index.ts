export * from './store';
export * from './theme';
