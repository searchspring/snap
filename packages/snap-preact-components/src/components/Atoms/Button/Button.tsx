/** @jsx jsx */
/** @jsx h */

import { h, Fragment, ComponentChildren } from 'preact';

import { jsx, css } from '@emotion/react';
import classnames from 'classnames';
import { observer } from 'mobx-react-lite';

import { ComponentProps, StylingCSS } from '../../../types';
import { Theme, useTheme, CacheProvider } from '../../../providers';
import { useA11y } from '../../../hooks/useA11y';
import { defined, mergeProps } from '../../../utilities';
import { Icon, IconProps } from '../Icon';

const CSS = {
	button: ({ color, backgroundColor, borderColor, theme }: ButtonProps) =>
		css({
			display: 'inline-flex',
			padding: '5px 10px',
			position: 'relative',
			color: color || theme?.colors?.primary,
			outline: 0,
			backgroundColor: backgroundColor || '#fff',
			border: `1px solid ${borderColor || color || theme?.colors?.primary || '#333'}`,
			'&:hover': {
				cursor: 'pointer',
				backgroundColor: theme?.colors?.hover || '#f8f8f8',
			},
			'&.ss__button--disabled': {
				opacity: 0.7,
				borderColor: 'rgba(51,51,51,0.7)',
				backgroundColor: 'initial',
				'&:hover': {
					cursor: 'default',
				},
			},
		}),
	native: () => css({}),
};

export const Button = observer((properties: ButtonProps): JSX.Element => {
	const globalTheme: Theme = useTheme();
	const theme = { ...globalTheme, ...properties.theme };
	const defaultProps = {
		disableA11y: false,
	};

	const props = mergeProps('button', globalTheme, defaultProps, properties);

	const {
		backgroundColor,
		borderColor,
		color,
		content,
		children,
		disabled,
		native,
		onClick,
		disableA11y,
		disableStyles,
		className,
		icon,
		style,
		styleScript,
	} = props;

	const subProps: ButtonSubProps = {
		icon: {
			// default props
			// global theme
			...globalTheme?.components?.icon,
			// inherited props
			...defined({
				disableStyles,
			}),
			// component theme overrides
			theme: props?.theme,
		},
	};

	const styling: { css?: StylingCSS } = {};
	const stylingProps = { backgroundColor, borderColor, color, disabled, native, theme };

	if (!disableStyles) {
		if (native) {
			styling.css = [CSS.native(), style];
		} else {
			styling.css = [CSS.button(stylingProps), style];
		}

		// add styleScript to styling
		if (styleScript) {
			styling.css = styling.css || [];
			styling.css.push(styleScript(stylingProps));
		}
	} else if (style) {
		styling.css = [style];
	}

	const elementProps = {
		...styling,
		className: classnames('ss__button', { 'ss__button--disabled': disabled }, className),
		disabled,
		onClick: (e: React.MouseEvent<HTMLElement, MouseEvent>) => !disabled && onClick && onClick(e),
	};

	const a11yProps = {
		ref: (e: any) => useA11y(e),
	};

	return content || children || icon ? (
		<CacheProvider>
			{native ? (
				<button {...elementProps}>
					{content}
					{children}
					{icon && <Icon icon={icon} {...subProps.icon} />}
				</button>
			) : (
				<div {...(!disableA11y ? a11yProps : {})} {...elementProps} role={'button'} aria-disabled={disabled}>
					{content}
					{children}
					{icon && <Icon icon={icon} {...subProps.icon} />}
				</div>
			)}
		</CacheProvider>
	) : (
		<Fragment></Fragment>
	);
});

interface ButtonSubProps {
	icon: Partial<IconProps>;
}

export interface ButtonProps extends ComponentProps {
	backgroundColor?: string;
	borderColor?: string;
	color?: string;
	icon?: string;
	content?: string | JSX.Element;
	children?: ComponentChildren;
	disabled?: boolean;
	native?: boolean;
	onClick?: (e: React.MouseEvent<HTMLElement, MouseEvent>) => void;
	disableA11y?: boolean;
}
