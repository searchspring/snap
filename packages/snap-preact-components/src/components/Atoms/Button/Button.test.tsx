import { h } from 'preact';

import { render } from '@testing-library/preact';
import userEvent from '@testing-library/user-event';

import { Button } from './Button';
import { ThemeProvider } from '../../../providers/theme';

describe('Button Component', () => {
	describe('styled', () => {
		it('does not render without children or content props', () => {
			const rendered = render(<Button />);
			expect(rendered.container).toBeInTheDocument();

			const buttonElement = rendered.container.querySelector('.ss-button');
			expect(buttonElement).not.toBeInTheDocument();
		});

		it('renders with content prop', () => {
			const content = 'button1';
			const rendered = render(<Button content={content} />);

			const buttonElement = rendered.container.querySelector('.ss-button');
			expect(buttonElement).toBeInTheDocument();

			expect(buttonElement.classList.length).toBeGreaterThan(1);

			const buttonElementByContent = rendered.getByText(content);
			expect(buttonElementByContent).toBeInTheDocument();
		});

		it('renders with children prop', () => {
			const content = <h1 class="child">childbutton</h1>;
			const rendered = render(<Button>{content}</Button>);

			const buttonElement = rendered.container.querySelector('.ss-button');
			expect(buttonElement).toBeInTheDocument();

			const childElement = buttonElement.querySelector('.child');
			expect(childElement).toBeInTheDocument();

			const childElementAltSelector = rendered.getByText('childbutton');
			expect(childElementAltSelector).toBeInTheDocument();
		});

		it('renders with children and content props', () => {
			const content = 'contenttext';
			const childrenContent = 'buttontext';
			const rendered = render(<Button content={content}>{childrenContent}</Button>);

			const buttonElement = rendered.getByText(content + childrenContent);
			expect(buttonElement).toBeInTheDocument();
		});

		it('renders with className prop', () => {
			const className = 'classy';
			const content = 'classy button';

			const rendered = render(<Button className={className}>{content}</Button>);

			const button = rendered.container.querySelector(`.${className}`);
			expect(button).toBeInTheDocument();
		});

		it('renders with colors prop', () => {
			const content = 'colorful button';
			const color = 'blue';
			const backgroundColor = 'green';
			const borderColor = 'red';

			const rendered = render(
				<Button color={color} backgroundColor={backgroundColor} borderColor={borderColor}>
					{content}
				</Button>
			);

			const button = rendered.getByText(content);

			const styles = getComputedStyle(button);
			expect(styles.color).toBe(color);
			expect(styles.backgroundColor).toBe(backgroundColor);
			expect(styles.borderColor).toBe(borderColor);
			expect(button).toBeInTheDocument();
		});

		it('fires onClick prop when clicked', () => {
			const clickFn = jest.fn();
			const content = 'clickable button';

			const rendered = render(<Button onClick={clickFn}>{content}</Button>);

			const buttonElement = rendered.container.querySelector('.ss-button');
			userEvent.click(buttonElement);

			expect(clickFn).toHaveBeenCalled();
		});

		it('adds class "ss-button__disabled" and prevents onClick when disabled by prop', () => {
			const clickFn = jest.fn();
			const content = 'disabled button';

			const rendered = render(<Button onClick={clickFn} disabled content={content} />);

			const buttonElement = rendered.container.querySelector(`.ss-button.ss-button__disabled`);
			expect(buttonElement).toBeInTheDocument();

			userEvent.click(buttonElement);
			expect(clickFn).not.toHaveBeenCalled();
		});

		it('renders with additional style using prop', () => {
			const content = 'styled button';
			const style = {
				padding: '20px',
			};

			const rendered = render(<Button style={style} content={content} />);
			const buttonElement = rendered.container.querySelector('.ss-button');
			const styles = getComputedStyle(buttonElement);

			expect(styles.padding).toBe(style.padding);
		});

		it('can disable styles', () => {
			const content = 'styleless button';

			const rendered = render(<Button disableStyles>{content}</Button>);

			const buttonElement = rendered.container.querySelector('.ss-button');
			expect(buttonElement.classList.length).toBe(2);
			expect(buttonElement.classList[1]).toMatch(/^css-0/);
			expect(buttonElement).toBeInTheDocument();
		});

		it('is themeable with ThemeProvider', () => {
			const content = 'themed button';
			const globalTheme = {
				components: {
					button: {
						color: 'blue',
					},
				},
			};

			const rendered = render(
				<ThemeProvider theme={globalTheme}>
					<Button content={content} />
				</ThemeProvider>
			);

			const buttonElement = rendered.container.querySelector('.ss-button');

			const styles = getComputedStyle(buttonElement);

			expect(styles.color).toBe(globalTheme.components.button.color);
			expect(buttonElement).toBeInTheDocument();
		});

		it('is themeable with theme prop', () => {
			const content = 'themed button';
			const propTheme = {
				components: {
					button: {
						color: 'blue',
					},
				},
			};

			const rendered = render(<Button content={content} theme={propTheme} />);

			const buttonElement = rendered.container.querySelector('.ss-button');

			const styles = getComputedStyle(buttonElement);

			expect(styles.color).toBe(propTheme.components.button.color);
			expect(buttonElement).toBeInTheDocument();
		});

		it('is themeable and theme prop overrides ThemeProvider', () => {
			const content = 'themed button';
			const globalTheme = {
				components: {
					button: {
						color: 'blue',
					},
				},
			};

			const propTheme = {
				components: {
					button: {
						color: 'red',
					},
				},
			};

			const rendered = render(
				<ThemeProvider theme={globalTheme}>
					<Button theme={propTheme} content={content} />
				</ThemeProvider>
			);

			const buttonElement = rendered.container.querySelector('.ss-button');

			const styles = getComputedStyle(buttonElement);

			expect(styles.color).toBe(propTheme.components.button.color);
			expect(buttonElement).toBeInTheDocument();
		});
	});

	describe('native', () => {
		it('does not render without children or content props', () => {
			const rendered = render(<Button native />);
			expect(rendered.container).toBeInTheDocument();

			const buttonElement = rendered.container.querySelector('.ss-button');
			expect(buttonElement).not.toBeInTheDocument();
		});

		it('renders with content prop', () => {
			const content = 'button1';
			const rendered = render(<Button native content={content} />);

			const buttonElement = rendered.container.querySelector('.ss-button');
			expect(buttonElement).toBeInTheDocument();

			expect(buttonElement.classList.length).toBe(2);

			const buttonElementByContent = rendered.getByText(content);
			expect(buttonElementByContent).toBeInTheDocument();
		});

		it('renders with children prop', () => {
			const content = <h1 class="child">childbutton</h1>;
			const rendered = render(<Button native>{content}</Button>);

			const buttonElement = rendered.container.querySelector('.ss-button');
			expect(buttonElement).toBeInTheDocument();

			const childElement = buttonElement.querySelector('.child');
			expect(childElement).toBeInTheDocument();

			const childElementAltSelector = rendered.getByText('childbutton');
			expect(childElementAltSelector).toBeInTheDocument();
		});

		it('renders with children and content props', () => {
			const content = 'contenttext';
			const childrenContent = 'buttontext';
			const rendered = render(
				<Button native content={content}>
					{childrenContent}
				</Button>
			);

			const buttonElement = rendered.container.querySelector('.ss-button');
			expect(buttonElement).toBeInTheDocument();

			const childElementAltSelector = rendered.getByText(content + childrenContent);
			expect(childElementAltSelector).toBeInTheDocument();
		});

		it('renders with className prop', () => {
			const className = 'classy';
			const content = 'classy button';

			const rendered = render(
				<Button native className={className}>
					{content}
				</Button>
			);

			const button = rendered.container.querySelector(`.${className}`);
			expect(button).toBeInTheDocument();
		});

		it('does not render with colors prop', () => {
			const content = 'colorful button';
			const color = 'blue';
			const backgroundColor = 'green';
			const borderColor = 'red';

			const rendered = render(
				<Button native color={color} backgroundColor={backgroundColor} borderColor={borderColor}>
					{content}
				</Button>
			);

			const button = rendered.getByText(content);

			const styles = getComputedStyle(button);
			expect(styles.color).not.toBe(color);
			expect(styles.backgroundColor).not.toBe(backgroundColor);
			expect(styles.borderColor).not.toBe(borderColor);
			expect(button).toBeInTheDocument();
		});

		it('fires onClick prop when clicked', () => {
			const clickFn = jest.fn();
			const content = 'clickable button';

			const rendered = render(
				<Button native onClick={clickFn}>
					{content}
				</Button>
			);

			const buttonElement = rendered.container.querySelector('.ss-button');
			userEvent.click(buttonElement);

			expect(clickFn).toHaveBeenCalled();
		});

		it('adds class "ss-button__disabled" and prevents onClick when disabled by prop', () => {
			const clickFn = jest.fn();
			const content = 'disabled button';

			const rendered = render(<Button native onClick={clickFn} disabled content={content} />);

			const button = rendered.container.querySelector(`.ss-button.ss-button__disabled`);
			expect(button).toBeInTheDocument();

			userEvent.click(button);
			expect(clickFn).not.toHaveBeenCalled();
		});
	});
});
