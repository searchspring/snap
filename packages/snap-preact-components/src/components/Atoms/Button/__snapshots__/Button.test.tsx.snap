// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component native Can enable/disable useAlly with disableA11y prop 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  padding: 5px 10px;
  position: relative;
  outline: 0;
  background-color: #fff;
  border: 1px solid #333;
}

.emotion-0:hover {
  cursor: pointer;
  background-color: #f8f8f8;
}

.emotion-0.ss__button--disabled {
  opacity: 0.7;
  border-color: rgba(51,51,51,0.7);
  background-color: initial;
}

.emotion-0.ss__button--disabled:hover {
  cursor: default;
}

<div
    class="ss__button emotion-0"
    role="button"
    ssa11y="true"
    tabindex="0"
  >
    button1
  </div>
</DocumentFragment>
`;

exports[`Button Component native Can enable/disable useAlly with disableA11y prop 2`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  padding: 5px 10px;
  position: relative;
  outline: 0;
  background-color: #fff;
  border: 1px solid #333;
}

.emotion-0:hover {
  cursor: pointer;
  background-color: #f8f8f8;
}

.emotion-0.ss__button--disabled {
  opacity: 0.7;
  border-color: rgba(51,51,51,0.7);
  background-color: initial;
}

.emotion-0.ss__button--disabled:hover {
  cursor: default;
}

<div
    class="ss__button emotion-0"
    role="button"
  >
    button1
  </div>
</DocumentFragment>
`;

exports[`Button Component native adds class "ss__button--disabled" and prevents onClick when disabled by prop 1`] = `
<DocumentFragment>
  <button
    class="ss__button ss__button--disabled emotion-0"
    disabled=""
  >
    disabled button
  </button>
</DocumentFragment>
`;

exports[`Button Component native adds class "ss__button--disabled" and prevents onClick when disabled by prop 2`] = `
<DocumentFragment>
  <button
    class="ss__button ss__button--disabled emotion-0"
    disabled=""
  >
    disabled button
  </button>
</DocumentFragment>
`;

exports[`Button Component native does not render with colors prop 1`] = `
<DocumentFragment>
  <button
    class="ss__button emotion-0"
  >
    colorful button
  </button>
</DocumentFragment>
`;

exports[`Button Component native does not render without children or content props 1`] = `<DocumentFragment />`;

exports[`Button Component native fires onClick prop when clicked 1`] = `
<DocumentFragment>
  <button
    class="ss__button emotion-0"
  >
    clickable button
  </button>
</DocumentFragment>
`;

exports[`Button Component native fires onClick prop when clicked 2`] = `
<DocumentFragment>
  <button
    class="ss__button emotion-0"
  >
    clickable button
  </button>
</DocumentFragment>
`;

exports[`Button Component native renders with children and content props 1`] = `
<DocumentFragment>
  <button
    class="ss__button emotion-0"
  >
    contenttextbuttontext
  </button>
</DocumentFragment>
`;

exports[`Button Component native renders with children prop 1`] = `
<DocumentFragment>
  <button
    class="ss__button emotion-0"
  >
    <h1
      class="child"
    >
      childbutton
    </h1>
  </button>
</DocumentFragment>
`;

exports[`Button Component native renders with className prop 1`] = `
<DocumentFragment>
  <button
    class="ss__button classy emotion-0"
  >
    classy button
  </button>
</DocumentFragment>
`;

exports[`Button Component native renders with content prop 1`] = `
<DocumentFragment>
  <button
    class="ss__button emotion-0"
  >
    button1
  </button>
</DocumentFragment>
`;

exports[`Button Component native renders with icon prop 1`] = `
<DocumentFragment>
  .emotion-1 {
  width: 16px;
  height: 16px;
  position: relative;
}

<button
    class="ss__button emotion-0"
  >
    colorful button
    <svg
      class="ss__icon ss__icon--close-thin emotion-1"
      viewBox="0 0 56 56"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M56 5.638l-22.362 22.362 22.362 22.362-5.638 5.638-22.362-22.362-22.362 22.362-5.638-5.638 22.362-22.362-22.362-22.362 5.638-5.638 22.362 22.362 22.362-22.362z"
      />
    </svg>
  </button>
</DocumentFragment>
`;

exports[`Button Component styled adds class "ss__button--disabled" and prevents onClick when disabled by prop 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  padding: 5px 10px;
  position: relative;
  outline: 0;
  background-color: #fff;
  border: 1px solid #333;
}

.emotion-0:hover {
  cursor: pointer;
  background-color: #f8f8f8;
}

.emotion-0.ss__button--disabled {
  opacity: 0.7;
  border-color: rgba(51,51,51,0.7);
  background-color: initial;
}

.emotion-0.ss__button--disabled:hover {
  cursor: default;
}

<div
    aria-disabled="true"
    class="ss__button ss__button--disabled emotion-0"
    disabled="true"
    role="button"
    ssa11y="true"
    tabindex="0"
  >
    disabled button
  </div>
</DocumentFragment>
`;

exports[`Button Component styled adds class "ss__button--disabled" and prevents onClick when disabled by prop 2`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  padding: 5px 10px;
  position: relative;
  outline: 0;
  background-color: #fff;
  border: 1px solid #333;
}

.emotion-0:hover {
  cursor: pointer;
  background-color: #f8f8f8;
}

.emotion-0.ss__button--disabled {
  opacity: 0.7;
  border-color: rgba(51,51,51,0.7);
  background-color: initial;
}

.emotion-0.ss__button--disabled:hover {
  cursor: default;
}

<div
    aria-disabled="true"
    class="ss__button ss__button--disabled emotion-0"
    disabled="true"
    role="button"
    ssa11y="true"
    tabindex="0"
  >
    disabled button
  </div>
</DocumentFragment>
`;

exports[`Button Component styled can disable styles 1`] = `
<DocumentFragment>
  <div
    class="ss__button"
    role="button"
    ssa11y="true"
    tabindex="0"
  >
    styleless button
  </div>
</DocumentFragment>
`;

exports[`Button Component styled does not render without children or content props 1`] = `<DocumentFragment />`;

exports[`Button Component styled fires onClick prop when clicked 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  padding: 5px 10px;
  position: relative;
  outline: 0;
  background-color: #fff;
  border: 1px solid #333;
}

.emotion-0:hover {
  cursor: pointer;
  background-color: #f8f8f8;
}

.emotion-0.ss__button--disabled {
  opacity: 0.7;
  border-color: rgba(51,51,51,0.7);
  background-color: initial;
}

.emotion-0.ss__button--disabled:hover {
  cursor: default;
}

<div
    class="ss__button emotion-0"
    role="button"
    ssa11y="true"
    tabindex="0"
  >
    clickable button
  </div>
</DocumentFragment>
`;

exports[`Button Component styled fires onClick prop when clicked 2`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  padding: 5px 10px;
  position: relative;
  outline: 0;
  background-color: #fff;
  border: 1px solid #333;
}

.emotion-0:hover {
  cursor: pointer;
  background-color: #f8f8f8;
}

.emotion-0.ss__button--disabled {
  opacity: 0.7;
  border-color: rgba(51,51,51,0.7);
  background-color: initial;
}

.emotion-0.ss__button--disabled:hover {
  cursor: default;
}

<div
    class="ss__button emotion-0"
    role="button"
    ssa11y="true"
    tabindex="0"
  >
    clickable button
  </div>
</DocumentFragment>
`;

exports[`Button Component styled is themeable and theme prop overrides ThemeProvider 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  padding: 5px 10px;
  position: relative;
  color: red;
  outline: 0;
  background-color: #fff;
  border: 1px solid red;
}

.emotion-0:hover {
  cursor: pointer;
  background-color: #f8f8f8;
}

.emotion-0.ss__button--disabled {
  opacity: 0.7;
  border-color: rgba(51,51,51,0.7);
  background-color: initial;
}

.emotion-0.ss__button--disabled:hover {
  cursor: default;
}

<div
    class="ss__button emotion-0"
    role="button"
    ssa11y="true"
    tabindex="0"
  >
    themed button
  </div>
</DocumentFragment>
`;

exports[`Button Component styled is themeable with ThemeProvider 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  padding: 5px 10px;
  position: relative;
  color: blue;
  outline: 0;
  background-color: #fff;
  border: 1px solid blue;
}

.emotion-0:hover {
  cursor: pointer;
  background-color: #f8f8f8;
}

.emotion-0.ss__button--disabled {
  opacity: 0.7;
  border-color: rgba(51,51,51,0.7);
  background-color: initial;
}

.emotion-0.ss__button--disabled:hover {
  cursor: default;
}

<div
    class="ss__button emotion-0"
    role="button"
    ssa11y="true"
    tabindex="0"
  >
    themed button
  </div>
</DocumentFragment>
`;

exports[`Button Component styled is themeable with theme prop 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  padding: 5px 10px;
  position: relative;
  color: blue;
  outline: 0;
  background-color: #fff;
  border: 1px solid blue;
}

.emotion-0:hover {
  cursor: pointer;
  background-color: #f8f8f8;
}

.emotion-0.ss__button--disabled {
  opacity: 0.7;
  border-color: rgba(51,51,51,0.7);
  background-color: initial;
}

.emotion-0.ss__button--disabled:hover {
  cursor: default;
}

<div
    class="ss__button emotion-0"
    role="button"
    ssa11y="true"
    tabindex="0"
  >
    themed button
  </div>
</DocumentFragment>
`;

exports[`Button Component styled renders with additional style using prop 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  padding: 5px 10px;
  position: relative;
  outline: 0;
  background-color: #fff;
  border: 1px solid #333;
  padding: 20px;
}

.emotion-0:hover {
  cursor: pointer;
  background-color: #f8f8f8;
}

.emotion-0.ss__button--disabled {
  opacity: 0.7;
  border-color: rgba(51,51,51,0.7);
  background-color: initial;
}

.emotion-0.ss__button--disabled:hover {
  cursor: default;
}

<div
    class="ss__button emotion-0"
    role="button"
    ssa11y="true"
    tabindex="0"
  >
    styled button
  </div>
</DocumentFragment>
`;

exports[`Button Component styled renders with children and content props 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  padding: 5px 10px;
  position: relative;
  outline: 0;
  background-color: #fff;
  border: 1px solid #333;
}

.emotion-0:hover {
  cursor: pointer;
  background-color: #f8f8f8;
}

.emotion-0.ss__button--disabled {
  opacity: 0.7;
  border-color: rgba(51,51,51,0.7);
  background-color: initial;
}

.emotion-0.ss__button--disabled:hover {
  cursor: default;
}

<div
    class="ss__button emotion-0"
    role="button"
    ssa11y="true"
    tabindex="0"
  >
    contenttextbuttontext
  </div>
</DocumentFragment>
`;

exports[`Button Component styled renders with children prop 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  padding: 5px 10px;
  position: relative;
  outline: 0;
  background-color: #fff;
  border: 1px solid #333;
}

.emotion-0:hover {
  cursor: pointer;
  background-color: #f8f8f8;
}

.emotion-0.ss__button--disabled {
  opacity: 0.7;
  border-color: rgba(51,51,51,0.7);
  background-color: initial;
}

.emotion-0.ss__button--disabled:hover {
  cursor: default;
}

<div
    class="ss__button emotion-0"
    role="button"
    ssa11y="true"
    tabindex="0"
  >
    <h1
      class="child"
    >
      childbutton
    </h1>
  </div>
</DocumentFragment>
`;

exports[`Button Component styled renders with className prop 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  padding: 5px 10px;
  position: relative;
  outline: 0;
  background-color: #fff;
  border: 1px solid #333;
}

.emotion-0:hover {
  cursor: pointer;
  background-color: #f8f8f8;
}

.emotion-0.ss__button--disabled {
  opacity: 0.7;
  border-color: rgba(51,51,51,0.7);
  background-color: initial;
}

.emotion-0.ss__button--disabled:hover {
  cursor: default;
}

<div
    class="ss__button classy emotion-0"
    role="button"
    ssa11y="true"
    tabindex="0"
  >
    classy button
  </div>
</DocumentFragment>
`;

exports[`Button Component styled renders with colors prop 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  padding: 5px 10px;
  position: relative;
  color: blue;
  outline: 0;
  background-color: green;
  border: 1px solid red;
}

.emotion-0:hover {
  cursor: pointer;
  background-color: #f8f8f8;
}

.emotion-0.ss__button--disabled {
  opacity: 0.7;
  border-color: rgba(51,51,51,0.7);
  background-color: initial;
}

.emotion-0.ss__button--disabled:hover {
  cursor: default;
}

<div
    class="colortest"
  >
    <button>
      a dummy button
    </button>
    <div
      class="ss__button emotion-0"
      role="button"
      ssa11y="true"
      tabindex="0"
    >
      colorful button
    </div>
  </div>
</DocumentFragment>
`;

exports[`Button Component styled renders with content prop 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  padding: 5px 10px;
  position: relative;
  outline: 0;
  background-color: #fff;
  border: 1px solid #333;
}

.emotion-0:hover {
  cursor: pointer;
  background-color: #f8f8f8;
}

.emotion-0.ss__button--disabled {
  opacity: 0.7;
  border-color: rgba(51,51,51,0.7);
  background-color: initial;
}

.emotion-0.ss__button--disabled:hover {
  cursor: default;
}

<div
    class="ss__button emotion-0"
    role="button"
    ssa11y="true"
    tabindex="0"
  >
    button1
  </div>
</DocumentFragment>
`;

exports[`Button Component styled renders with icon prop 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  padding: 5px 10px;
  position: relative;
  outline: 0;
  background-color: #fff;
  border: 1px solid #333;
}

.emotion-0:hover {
  cursor: pointer;
  background-color: #f8f8f8;
}

.emotion-0.ss__button--disabled {
  opacity: 0.7;
  border-color: rgba(51,51,51,0.7);
  background-color: initial;
}

.emotion-0.ss__button--disabled:hover {
  cursor: default;
}

.emotion-1 {
  width: 16px;
  height: 16px;
  position: relative;
}

<div
    class="ss__button emotion-0"
    role="button"
    ssa11y="true"
    tabindex="0"
  >
    colorful button
    <svg
      class="ss__icon ss__icon--close-thin emotion-1"
      viewBox="0 0 56 56"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M56 5.638l-22.362 22.362 22.362 22.362-5.638 5.638-22.362-22.362-22.362 22.362-5.638-5.638 22.362-22.362-22.362-22.362 5.638-5.638 22.362 22.362 22.362-22.362z"
      />
    </svg>
  </div>
</DocumentFragment>
`;

exports[`Button Component uses bocachica theme 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  padding: 5px 10px;
  position: relative;
  color: #6d7175;
  outline: 0;
  background-color: #fff;
  border: 1px solid #6d7175;
  border-radius: 3px;
}

.emotion-0:hover {
  cursor: pointer;
  background-color: #000000;
  color: #ffffff;
  border-color: #ffffff;
}

.emotion-0.ss__button--disabled {
  opacity: 0.7;
  border-color: rgba(51,51,51,0.7);
  background-color: initial;
}

.emotion-0.ss__button--disabled:hover {
  cursor: default;
}

<div
    class="ss__button emotion-0"
    role="button"
    ssa11y="true"
    tabindex="0"
  >
    hello world
  </div>
</DocumentFragment>
`;
